code/modules/integrated_electronics/subtypes/arithmetic.dm

```diff
@@ -288,11 +279,10 @@
 
 /obj/item/integrated_circuit/arithmetic/square_root/do_work()
 	var/result = 0
-	for(var/k in 1 to inputs.len)
-		var/datum/integrated_io/I = inputs[k]
-		I.pull_data()
-		if(isnum(I.data))
-			result = sqrt(I.data)
+	for(var/k in 2 to inputs.len)
+		var/I = get_pin_data(IC_INPUT, k)
+		if(isnum(I))
+			result = sqrt(I)
```

Comment:
Result is reset each iteration of the loop and previous values mean nothing so you should step backwards instead
```
for(var/k in inputs.len to 2 step -1)
   var/I = get_pin_data(IC_INPUT, k)
   if(isnum(I)
      result = sqrt(I)
      break
```

Reply:
I missed there +=